subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/TransformationSubject.java

```diff
@@ -36,27 +37,33 @@ public static TransformationSubject initial(File file) {
         return new InitialFileTransformationSubject(file);
     }
 
-    public static TransformationSubject initial(ComponentArtifactIdentifier artifactId, File file) {
-        return new InitialArtifactTransformationSubject(artifactId, file);
+    public static TransformationSubject initial(ComponentArtifactIdentifier artifactId, File file, ArtifactTransformDependencies dependencies) {
+        return new InitialArtifactTransformationSubject(artifactId, file, dependencies);
     }
 
     /**
      * The files which should be transformed.
      */
     public abstract ImmutableList<File> getFiles();
 
+    /**
+     * Gives access to the artifacts of the dependencies of the subject of the transformation
+     */
+    public abstract ArtifactTransformDependencies getDependencies();
```

Comment:
:o: `getArtifactDependencies()`?

Reply:
We have a plan to get rid of `TransformationSubject` altogether. Let's implement that instead.